### DISCUSSION TITLE
Don't working touchpad

### DISCUSSION THREAD

**Discussion by Coconurminz** [Qonfused/ASUS-ZenBook-Duo-14-UX481-Hackintosh#50] [Post-Install Help]
Hello! After installing MacOS Ventura, the touchpad does not work, although the project specifies full touchpad support. What can be done to make it work?

**Comment by Qonfused** [comment]
The trackpad should work if it is properly GPIO pinned, but it may be timing out when falling back to I2C interrupts.

Essentially what this means is that instead of using GPIO -- an event-based system (where messages are only sent to and from the CPU as needed) -- instead, the trackpad driver may be falling back to polling messages from the trackpad at a regular interval.

If the trackpad is somehow misconfigured, polling over I2C will not receive any responses from the trackpad. You can check for this behavior with `sudo dmesg | grep -i Voodoo` to spot any messages like these:
```
[  155.146947]: VoodooI2CControllerDriver::pci8086,2eb I2C Transaction error details
[  155.146967]: VoodooI2CControllerDriver::pci8086,2eb slave address not acknowledged (7bit mode)
[  155.146971]: VoodooI2CControllerDriver::pci8086,2eb I2C Transaction error: 0x0b800001 - aborting
```

You'll want to add `msgbuf=1048576` to your config.plist's boot-args (under `NVRAM -> Add -> 7C436110-AB2A-4BBB-A880-FE41995C9F82`), which will expand the buffer size for dmesg. This will prevent logs from being truncated, as the default size is very small and is likely to be overwritten very quickly.

You'll also want to grab some more detailed information about what hardware is detected for the VoodooI2C drivers by running the below command ([example output](https://github.com/Qonfused/VoodooInjector#finding-product-and-vendor-ids)):
```sh
ioreg -rxn IOHIDInterface -k "VoodooI2CServices Supported" -t | grep -E 'IOACPIPlatformDevice|IOPCIDevice|VoodooI2CDeviceNub|VendorID|ProductID|HIDEventDriver  <class VoodooI2C|Interface  <class VoodooI2C' | cut -d "<" -f1
```

**Reply by Qonfused** [reply]
You can also try closing the lid of the laptop for a few seconds and then waking it. This will force the trackpad and driver to re-initialize.

This is not a fix, though there have been some intermittent issues resolved in the past (e.g. #24), which might suggest that your trackpad is falling back to I2C polling.

**Reply by Coconurminz** [reply]
When you enter the commands that you gave, VoodooI2C is nowhere to be found, and when you enter the
`ioreg -rxn IOHIDInterface -k "VoodooI2CServices Supported" -t | grep -E 'IOACPIPlatformDevice|IOPCIDevice|VoodooI2CDeviceNub|VendorID|ProductID|HIDEventDriver  <class VoodooI2C|Interface  <class VoodooI2C' | cut -d "<" -f1`
 command there is just empty output
